sdk/tables/azure-data-tables/azure/data/tables/_entity.py

```diff
@@ -22,22 +22,26 @@ class TableEntity(dict):
 
     """
 
+    _metadata = {}
+
+
     def _set_metadata(self):
         if "Timestamp" in self.keys():
-            self._metadata = {  # pylint:disable=W0201
+            TableEntity._metadata = {  # pylint:disable=W0201
```

Comment:
This has to be `self._metadata` - otherwise we are setting the same metadata for all instances of TableEntity

Reply:
If this is changed to `self._metadata` then `_metadata` will show up in the keys because of the inheritance from `dict`. Is this the behavior we are expecting?

Reply:
Hah interesting - I didn't realize that - Maybe we should rethink the `dict` inheritance - it seems to be tying our hands a bit.